cmake/EthOptions.cmake

```diff
@@ -1,7 +1,9 @@
 macro(configure_project)
 	# Default to RelWithDebInfo configuration if no configuration is explicitly specified.
-	set(CMAKE_BUILD_TYPE RelWithDebInfo CACHE STRING "Build type on single-configuration generators")
-
+	if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
```

Comment:
I think this should read
`	if(NOT CMAKE_BUILD_TYPE AND CMAKE_CONFIGURATION_TYPES)`
I noticed the tests were still slow and found the variable still set - because CMAKE_CONFIGURATION_TYPES was set to "Debug;Release;RelWithDebInfo", which seemed no reason to not set the build type. Rather, having no available configuration types would be a reason to not set the build type.  @chfast ?